### PR TITLE
Support BigDecimal comparison with and initialization from BigRational

### PR DESCRIPTION
**Pull request by Sija**
This PR allows below code to work:

```cr
def add_numbers(*numbers : Number)
  numbers.reduce(0.to_big_d) { |sum, n| sum + n.to_big_d }
end

add_numbers(1, 1.23, 3.to_big_d, 3.3.to_big_r) # => 8.53
```

### CODE REVIEW

File: src/big/big_decimal.cr [hunk 1]
```diff
@@ -133,6 +134,11 @@ struct BigDecimal < Number
     initialize(num.to_s)
   end
 
+  # Creates a new `BigDecimal` from `BigRational`.
+  def initialize(num : BigRational)
+    initialize(num.to_f64)
```
Comment (RX14) [inline src/big/big_decimal.cr]: How about `num.numerator.to_big_d / num.denominator.to_big_d`?

Comment (Sija) [reply src/big/big_decimal.cr]: done & force-pushed

File: src/big/big_decimal.cr [hunk 2]
```diff
@@ -133,6 +134,13 @@ struct BigDecimal < Number
     initialize(num.to_s)
   end
 
+  # Creates a new `BigDecimal` from `BigRational`.
+  def initialize(num : BigRational)
```
Comment (RX14) [inline src/big/big_decimal.cr]: but why not make this a `self.new` then you can just return the original number instead of copying the value and scale...

Comment (Sija) [reply src/big/big_decimal.cr]: True indeed, I've forgot about that, thx! force-pushed again

**Comment by Sija** [comment]
Is it GTG? Could we have a 2nd review here?

**Pull request merged**